doc/OnlineDocs/contributed_packages/sensitivity_toolbox.rst

```diff
@@ -0,0 +1,187 @@
+Sensitivity Toolbox
+===================
+
+The sensitivity toolbox provides a Pyomo interface to sIPOPT to very quickly computer approximate solutions to nonlinear programs with a small perturbation in model parameters. See the `sIPOPT documentation <https://projects.coin-or.org/Ipopt/wiki/sIpopt>`_ or the `following paper <https://link.springer.com/article/10.1007/s12532-012-0043-2>`_ for additional details:
+
+  H. Pirnay, R. Lopez-Negrete, and L.T. Biegler, Optimal Sensitivity based on IPOPT, Math. Prog. Comp., 4(4):307--331, 2012. 
+
+
+Using the Sensitivity Toolbox
+-----------------------------
+
+We will start with a motivating example:
+
+.. math::
+  \begin{align*}
+  \min_{x_1,x_2,x_3} \quad & x_1^2 + x_2^2 + x_3^2 \\
+  \mathrm{s.t.} \qquad & 6 x_1 + 3 x_2 + 2 x_3 - p_1 = 0 \\
+  & p_2 x_1 + x_2 - x_3 - 1 = 0 \\
+  & x_1, x_2, x_3 \geq 0
+  \end{align*}
+
+Here :math:`x_1`, :math:`x_2`, and :math:`x_3` are the decision variables while :math:`p_1` and :math:`p_2` are parameters. At first, let's consider :math:`p_1 = 4.5` and :math:`p_2 = 1.0`. Below is the model implemented in Pyomo.
+
+.. testcode::
```

Comment:
Why `.. testcode::` and not `.. doctests::`?

Reply:
I thought ``.. testcode::`` looked better, but now I like ``.. doctests::``.